extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriLightHandler.java

```diff
@@ -0,0 +1,322 @@
+                } else if (command instanceof IncreaseDecreaseType) {
+                    if (state != null) {
+                        int current = state.getBrightness().intValue();
+                        if (IncreaseDecreaseType.INCREASE.equals(command)) {
+                            setBrightness(new PercentType(Math.min(current + STEP, PercentType.HUNDRED.intValue())));
+                        } else {
+                            setBrightness(new PercentType(Math.max(current - STEP, PercentType.ZERO.intValue())));
+                        }
+                    } else {
+                        logger.debug("Cannot handle inc/dec as current state is not known.");
+                    }
+                } else {
+                    logger.debug("Cannot handle command {} for channel {}", command, channelUID);
+                }
+                break;
+            case CHANNEL_COLOR_TEMPERATURE:
+                if (command instanceof PercentType) {
+                    setColorTemperature((PercentType) command);
+                } else if (command instanceof IncreaseDecreaseType) {
+                    if (state != null) {
+                        int current = state.getColorTemperature().intValue();
+                        if (IncreaseDecreaseType.INCREASE.equals(command)) {
+                            setColorTemperature(
+                                    new PercentType(Math.min(current + STEP, PercentType.HUNDRED.intValue())));
+                        } else {
+                            setColorTemperature(new PercentType(Math.max(current - STEP, PercentType.ZERO.intValue())));
+                        }
+                    } else {
+                        logger.debug("Cannot handle inc/dec as current state is not known.");
+                    }
+                } else {
+                    logger.debug("Can't handle command {} on channel {}", command, channelUID);
+                }
+                break;
+            default:
+                logger.error("Unknown channel UID {}", channelUID);
+        }
+    }
+
+    /**
+     * This class is a Java wrapper for the raw JSON data about the light state.
+     */
+    private static class LightData {
+
+        private final Logger logger = LoggerFactory.getLogger(LightData.class);
+
+        JsonObject root;
+        JsonArray array;
+        JsonObject attributes;
+
+        public LightData() {
+            root = new JsonObject();
+            array = new JsonArray();
+            attributes = new JsonObject();
+            array.add(attributes);
+            root.add(LIGHT, array);
+        }
+
+        public LightData(JsonElement json) {
+            try {
+                root = json.getAsJsonObject();
+                array = root.getAsJsonArray(LIGHT);
+                attributes = array.get(0).getAsJsonObject();
+            } catch (JsonSyntaxException e) {
+                logger.error("JSON error: {}", e.getMessage(), e);
```

Comment:
Yes, I consider this situation here to be covered by [E.4.error](https://github.com/eclipse/smarthome/blob/master/docs/documentation/development/guidelines.md#e-logging): "system cannot function normally anymore, the user should report it as a severe bug". Would you agree?